decode/src/lib.rs

```diff
@@ -57,4 +57,14 @@ pub trait ParseFromModule: Sized {
 
     /// Parse the result of the above reads into `Self`.
     fn parse<'a>(id: Identifier, reads: impl Iterator<Item = &'a [u8]>) -> Result<Self, Error>;
+
+    /// Use `reads` and `parse` together to collect data from a buffer.
+    fn from_buf(id: Identifier, data: &[u8]) -> Result<Self, Error> {
```

Comment:
How is this being or going to be used? It looks like `data` needs to already have the entire memory map of the module in order for the offsets in the reads to correctly index it. Also, this doesn't take into account the page information in the reads, so I'm not sure how this would be used to parse that spans more than one page.

Reply:
I'm currently using this to parse EEPROM data made available by a Linux kernel driver as a file in sysfs.

https://github.com/oxidecomputer/x2/blob/860a02811c57fcff8fa5e281c8baf788ae944838/x2-evb/src/lib.rs#L43-L53

Reply:
I don't know the details of the Linux implementation there, but I don't understand how that can work. The datapath information is spread across many different pages, for example. How does the driver expose the idea of a page?